cmd/controller/main.go

```diff
@@ -53,5 +71,55 @@ func main() {
 		"reconciliation-timeout", reconciler.DefaultTimeout,
 		"The amount of time to give each reconciliation of a resource to complete before its context is canceled.")
 
-	sharedmain.MainWithContext(signals.NewContext(), "controller", ctors...)
```

Comment:
Can we have a second method in sharedmain where we pass a function to do:

```
if shouldEnableNetCertManagerController(ctx, client) {
		injection.Default.RegisterInformer(challenge.WithInformer)
		injection.Default.RegisterInformer(v1certificate.WithInformer)
		injection.Default.RegisterInformer(certificaterequest.WithInformer)
		injection.Default.RegisterInformer(clusterissuer.WithInformer)
		injection.Default.RegisterInformer(issuer.WithInformer)
		ctors = append(ctors, netcertmanager.NewController)
	}
```

to avoid duplicating code here?

Reply:
Or we just iterate on an array of informers. Moving to sharedmain should be the next iteration imho.